test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -22,43 +22,228 @@ import '../util.dart';
 import 'boilerplate_utilities_test.dart';
 
 void main() {
+  const publicPropsClassName = 'FooProps';
+  const propsClassName = '_\$$publicPropsClassName';
+  const publicStateClassName = 'FooState';
+  const stateClassName = '_\$$publicStateClassName';
+  const factoryDecl = '''
+      @Factory()
+      UiFactory<$publicPropsClassName> Foo =
+          // ignore: undefined_identifier
+          \$Foo;
+      ''';
+  const componentDecl = '''
+      @Component2()
+      class FooComponent extends UiComponent2<$publicPropsClassName> {}
+      ''';
+  const statefulComponentDecl = '''
+      @Component2()
+      class FooComponent extends UiStatefulComponent2<$publicPropsClassName, $publicStateClassName> {}
+      ''';
+
   group('AdvancedPropsAndStateClassMigrator', () {
+    var runCount = 0;
     final converter = ClassToMixinConverter();
-    final testSuggestor =
-        getSuggestorTester(AdvancedPropsAndStateClassMigrator(converter));
+    SuggestorTester testSuggestor({
+      bool convertClassesWithExternalSuperclass = false,
+      Map<String, String> visitedClassNames = const {},
+    }) {
+      runCount++;
+      final tester = getSuggestorTester(AdvancedPropsAndStateClassMigrator(
+          converter,
+          treatUnvisitedClassesAsExternal: runCount > 1,
```

Comment:
```suggestion
          treatUnvisitedClassesAsExternal: runCount % 2 == 0,
```
I might not be fully understanding this, but it looks like every time the codemod is run, the first `AdvancedPropsAndStateClassMigrator` has `treatUnvisitedClassesAsExternal` set to false and the second `AdvancedPropsAndStateClassMigrator` has `treatUnvisitedClassesAsExternal` set to true.  A couple tests below are simulating running the codemod twice so it might be more realistic to alternate setting `treatUnvisitedClassesAsExternal` to true and add an additional run of `AdvancedPropsAndStateClassMigrator` to the second codemod run. I don't know if it makes much of a difference in the tests, but just a thought.